### PR TITLE
Updates/strategy tweak v1

### PR DESCRIPTION
**Pull request by f2calv**
what do you think?

### CODE REVIEW

File: azure-pipelines.yml [hunk 1]
```diff
@@ -1,38 +1,41 @@
...
-      targetStage: 'linux-arm'
-      repository: 'myapp'
-      dockerfile: 'Dockerfile.Linux'
-  - template: yml-steps/build-and-push.yml
-    parameters:
-      targetStage: 'linux-x64'
-      repository: 'myapp'
-      dockerfile: 'Dockerfile.Linux'
+#need to add in these variables
+variables:
```
Comment (cadullms) [inline azure-pipelines.yml]: This looks as if people should be adding these variables through code, which they shouldn't. This should better be a hint at the doc in the readme.md.

File: azure-pipelines.yml [hunk 2]
```diff
@@ -1,38 +1,41 @@
 trigger:
-- master
+- updates/strategy-tweak
```
Comment (cadullms) [inline azure-pipelines.yml]: This should not go to master...

File: azure-pipelines.yml [hunk 3]
```diff
@@ -1,38 +1,42 @@
...
-    vmImage: 'windows-2019'
+    vmImage: $(vmImage)
   steps:
   - template: yml-steps/build-and-push.yml
     parameters:
-      targetStage: 'windows-x64'
+      targetStage: $(imageName)
       repository: 'myapp'
-      dockerfile: 'Dockerfile.Windows'
+      dockerfile: $(dockerfile)
```
Comment (cadullms) [inline azure-pipelines.yml:31]: This matrix thing is great... actually, it feels like one of the main scenarios for this feature. Many thanks for the hint!

File: azure-pipelines.yml (hunk 2 above)
Comment (cadullms) [reply azure-pipelines.yml]: Or at least, not with this specific branch. It should be at least a wildcard pattern, e.g. for topics/* or something like that.

**Comment by cadullms** [comment]
> what do you think?

Looks great, just a few things before approving. And didn't you mention something about some c# you changed?

File: azure-pipelines.yml (hunk 2 above)
Comment (f2calv) [reply azure-pipelines.yml]: I have reverted this trigger back to master

**Comment by f2calv** [comment]
> > what do you think?
> 
> Looks great, just a few things before approving. And didn't you mention something about some c# you changed?
I have updated the pr (I think).
Yes the other c# changes are in the az-cont-opts-demo

**Pull request merged**